src/backend/joanie/badges/providers/obf.py

```diff
@@ -0,0 +1,468 @@
+            logger.debug("Request URL was: %s", response.request.url)
+            logger.debug("Request body was: %s", response.request.body)
+            logger.debug("Response text was: %s", response.text)
+            raise AuthenticationError(msg)
+
+        return response.json().get("access_token")
+
+    @property
+    def _get_auth(self):
+        """Make access token generation dynamic."""
+
+        return OAuth2AccessToken(self._access_token)
+
+    def check_auth(self):
+        """Check OBF API credentials using the dedicated endpoint."""
+
+        url = f"/ping/{self.client_id}"
+        response = self.get(url)
+        if response.status_code != requests.codes.ok:  # pylint: disable=no-member
+            raise AuthenticationError("Invalid access token for OBF")
+        return response
+
+    @classmethod
+    def iter_json(cls, response: requests.Response) -> Iterable:
+        """Iterate over JSON lines serialization in API responses.
+
+        When multiple objects are returned by the API, they are streamed as
+        JSON lines instead of a JSON list, leading the response.json() method
+        to fail as it is expected a valid JSON list instead. We mitigate this
+        issue by forcing JSON serialization of each item in the response.
+        """
+
+        try:
+            json_ = response.json()
+            if isinstance(json_, list):
+                for item in json_:
+                    yield item
+            else:
+                yield json_
+        except requests.JSONDecodeError:
+            for line in response.iter_lines():
+                yield json.loads(line)
+
+    # pylint: disable=arguments-differ
+    def request(self, method, url, **kwargs):
+        """Make OBF API usage more developer-friendly:
+
+        - Automatically add the API root URL so that we can focus on the endpoints
+        - Automatically renew access token when expired
+        """
+
+        url = urljoin(self.api_root_url, f"{self.api_version_prefix}/{url}")
+        response = super().request(method, url, **kwargs)
+
+        # Try to regenerate the access token in case of 403 response
+        if (
+            response.status_code
+            == requests.codes.forbidden  # pylint: disable=no-member
+        ):
+            # Clear cached property and force access token update
+            del self._access_token
+            self.auth = self._get_auth
+            # Give it another try
+            return super().request(method, url, **kwargs)
+
+        # Check response status and raise an exception on error
+        try:
+            response.raise_for_status()
+        except requests.HTTPError as error:
+            logger.error(
+                "An error occurred while requesting URL: %s (status: %d)",
+                response.request.url,
+                response.status_code,
+            )
+            if self.raise_for_status:
+                raise error
```

Comment:
should be either logger.error or raise but not both?

Reply:
True. So I should just raise for status if asked to.